bullet_train-super_load_and_authorize_resource/test/controllers/concerns/bullet_train/loads_and_authorizes_resource_test.rb

```diff
@@ -1,23 +1,87 @@
 require "test_helper"
 
 class BulletTrain::LoadsAndAuthorizeResourceTest < ActiveSupport::TestCase
-  class TestClass
+  class TestControllerClass < ActionController::Base
     include BulletTrain::LoadsAndAuthorizesResource
+
+    attr_accessor :child_object, :current_user, :parent_object, :team
+
+    def self.regex_to_remove_controller_namespace
+    end
+
+    def can?(*args)
+    end
+  end
+
+  module Users
+    class TestControllerClass < ActionController::Base
+      include BulletTrain::LoadsAndAuthorizesResource
+
+      def self.regex_to_remove_controller_namespace
+        /^BulletTrain::LoadsAndAuthorizeResourceTest::/
+      end
+    end
   end
 
-  test "it defines .model_namespace_from_controller_namespace" do
-    assert TestClass.respond_to?(:model_namespace_from_controller_namespace)
+  test "model_namespace_from_controller_namespace returns an array of modules names based on the classes namespace minus regex_to_remove_controller_namespace" do
+    assert_equal ["BulletTrain", "LoadsAndAuthorizeResourceTest"], TestControllerClass.model_namespace_from_controller_namespace
+    assert_equal ["Users"], Users::TestControllerClass.model_namespace_from_controller_namespace
   end
 
   test "it defines .account_load_and_authorize_resource" do
-    assert TestClass.respond_to?(:account_load_and_authorize_resource)
+    assert TestControllerClass.respond_to?(:account_load_and_authorize_resource)
```

Comment:
There's also a specific assertion for respond tos we could use in the file:

```suggestion
    assert_respond_to TestControllerClass, :account_load_and_authorize_resource
```